tests/test_document.py

```diff
@@ -30,6 +30,75 @@ async def test_constructor_with_wrong_type_for_data(
         Document(database, "foo", data=cast(Any, doc))
 
 
+async def test_context_manager_by_creating_new_doc(database: Database) -> None:
+    from aiocouch.document import Document
+
+    new_doc_id = "new_doc"
+
+    # Use context manager to create & save new doc with some data in it
+    async with Document(database=database, id=new_doc_id) as document:
+        assert (len(document.keys())) == 1
+        assert document["_id"] == document.id == new_doc_id
+        document["king"] = "elvis"
+
+    # Verify whether the data was actually written to the server or not
+    saved_doc = await database.get(new_doc_id)
+
+    assert saved_doc["_id"] == saved_doc.id == new_doc_id
+    assert saved_doc.rev is not None
+    assert saved_doc.rev.startswith("1-")
+    assert "king" in saved_doc
+    assert saved_doc["king"] == "elvis"
+
+
+async def test_context_manager_by_retrieving_existing_doc(database: Database) -> None:
```

Comment:
Thanks for splitting the test. Though, I think this test should use on the predefined documents of the `filled_database` fixture: https://github.com/metricq/aiocouch/blob/master/tests/conftest.py#L91

Reply:
Updated the test to use the `filled_database` fixture.